documentation/Working_on_the_Bot.md

```diff
@@ -0,0 +1,126 @@
+ 1. If your PR is doing something like adding a new line or removing a new line, CODEOWNERS reserve the right to ask that you undo that change unless it was for a specific reason.  
+ 1. If you are adding a new command....**document**. Document the following things on help.json and the README.md  
+    1. The purpose of the command  
+    1. If the argument is called with any arguments.  
+       1. If it is called with any arguments, please either provide a good enough explanation of the arg that a user can tell what it will do before using the command. adding an example of how to call it with the args is not necessary but good practice.  
+ 1. If you are making a new Class of commands, add the class to bot.json following the convention already there.  
+ 1. Evidence of Testing. This one needs to be completed after the PR is opened. At that point, you will go on the channel on the CSSS Wall-E Test Server that was automatically created when the PR was opened and then test the [following functionality](#test-cases). Once you had done so, you can leave a comment on the PR stating that you had done the necessary testing.  
+ 1. Please provide ways to test whatever you just modified on the bot in the [Test Cases section below](#test-cases) so that future PRs can be tested to ensure they dont break *your code* when merging to master  
+ 
+ ## Test Cases  
+
+ 1. `.ping`  
+ 1. `.echo this is the test case`  
+ 1. `.help`  
+    1. Please ensure that the pagination is not effected by doing the following  
+       1. go to the last page and then hit next when on the last page to make sure it goes back to the beginning  
+       1. go to the last page from the first page by hitting previous  
+       1. make sure that the done emoji does delete the help output  
+1. `.here`  
+1. `.here k`  
+1. `.poll`  
+1. `.poll "question"`  
+1. `.poll question answer1 answer 2`  
+1. `.urban yo`  
+1. `.wolfram 1+1`  
+1. `.remindmein 20 seconds to test`  
+1. `.showreminders`  
+1. `.deletereminder <output from showreminder>`  
+1. `.remindmein 10 seconds to test`  
+   1. wait 10 seconds to ensure that the reminder did get sent
+1. `.iam <whatever_name_you_want_to_use>`  
+1. `.newrole <whatever_name_you_want_to_use>`  
+1. `.whois <whatever_name_you_want_to_use>`  
+1. `.iam <whatever_name_you_want_to_use>`  
+1. `.whois <whatever_name_you_want_to_use>`  
+1. `.deleterole <whatever_name_you_want_to_use>`  
+1. `.iamn <whatever_name_you_want_to_use>`  
+1. `.whois <whatever_name_you_want_to_use>`  
+1. `.deleterole <whatever_name_you_want_to_use>`  
+1. `.roles`  
+1. `.Roles`  
+
+ ## Reporting Issues  
+
+ If you come across issues that googling wont solve, you can email the bot-managers with the details at `csss-bot-manager@sfu.ca`. Please note that if your email is not detailed enough, the bots may not necessarily respond. Please over-provide rather than under-provide.

```

Comment:
Absolutely HARAM. We(At least not me) are not here to teach ppl how to python and bot, googling will solve and/or #projects-and-dev on our discord(maybe this'll bring the channel back from the dead).  At the very least I dont wanna trouble shoot over email, in person is much better.

Reply:
>Absolutely HARAM
Does that mean really good or really bad?

Reply:
bad

Reply:
what would you change it to?

Reply:
maybe say after googling first try out discord server and ask around in projects and dev channel, if that fails and only then as a last resort email the bot managers.